src/main/scala/org/apache/spark/sql/rikai/NDArray.scala

```diff
@@ -17,10 +17,46 @@
 package org.apache.spark.sql.rikai
 
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.catalyst.InternalRow
 
+/**
+  * NDArray type. Storing the persisted data from numpy ndarray.
+  *
+  * Note that this class does not support interoperation in scala yet.
+  * It will only show a array summary in "df.show()".
+  * But the data is accessible from Pyspark and Pytorch / Tensorflow readers.
+  *
+  * @todo consider to use arrow Array for interchangable data transfer.
+  *
+  * @param dtype
+  */
 @SQLUserDefinedType(udt = classOf[NDArrayType])
-class NDArray {
-  override def toString: String = s"ndarray()"
+class NDArray(val dtype: Byte) {
+
+  /**
+    * It will only display a summary using df.show().
+    */
+  override def toString: String = s"ndarray(${portableDtype}, ...)"
+
+  /**
+    * @see Python ``rikai.convert.PortableDataType``
+    */
+  def portableDtype: String = {
+    dtype match {
+      case 1  => "uint8"
+      case 2  => "uint16"
+      case 3  => "uint32"
+      case 4  => "uint64"
+      case 5  => "int8"
+      case 6  => "int16"
+      case 7  => "int32"
+      case 8  => "int64"
+      case 9  => "float"
+      case 10 => "double"
+      case 11 => "boolean"
```

Comment:
what if the ndarray isn't any of these types (https://numpy.org/devdocs/user/basics.types.html) or if it's a subtype (e.g., datetime64, timedelta64, etc)?

Reply:
what about make it a `str` value , i.e., `"float64"` for now?  Such values should be dictionary encoded by parquet later.

Reply:
sg